hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/OzoneKeyDetails.java

```diff
@@ -58,8 +56,8 @@ public OzoneKeyDetails(String volumeName, String bucketName, String keyName,
     super(volumeName, bucketName, keyName, size, creationTime,
         modificationTime, type, replicationFactor);
     this.ozoneKeyLocations = ozoneKeyLocations;
-    this.metadata = metadata;
     this.feInfo = feInfo;
+    this.getMetadata().putAll(metadata);
```

Comment:
Thanks @DaveTeng0  for this important patch to handle user defined metadata supplied through the s3 interface.
Setting the metadata member of the `OzoneKey` class that` OzoneKeyDetails` is extended from should be done by a setter method or a protected member.  Perhaps a setter method can be provided by the `OzoneKey` class to set the metadata member or the `OzoneKey` _private_ metadata member can be made accessible by changing to `protected`.

Reply:
oh!! good catch! let me add a setter method for it! Thanks Neils!